plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/binary/KSAnnotationDescriptorImpl.kt

```diff
@@ -65,10 +73,36 @@ private fun <T> ConstantValue<T>.toValue(): Any? = when (this) {
     else -> value
 }
 
-fun AnnotationDescriptor.createKSValueArguments(): List<KSValueArgument> =
-    allValueArguments.map { (name, constantValue) ->
+fun AnnotationDescriptor.createKSValueArguments(): List<KSValueArgument> {
+    val presentValueArguments = allValueArguments.map { (name, constantValue) ->
         KSValueArgumentLiteImpl.getCached(
             KSNameImpl.getCached(name.asString()),
             constantValue.toValue()
         )
     }
+    val presentValueArgumentNames = presentValueArguments.map { it.name.asString() }
+    val argumentsFromDefault = (this.type.constructor.declarationDescriptor as? ClassDescriptor)?.constructors?.single()?.let {
+        it.getAbsentDefaultArguments(presentValueArgumentNames)
+    } ?: emptyList()
+    return presentValueArguments.plus(argumentsFromDefault)
+}
+
+fun ClassConstructorDescriptor.getAbsentDefaultArguments(excludeNames: Collection<String>): Collection<KSValueArgument> {
+    return this.valueParameters.filterNot { param -> excludeNames.contains(param.name.asString()) || !param.hasDefaultValue() }
+        .map { param ->
+            KSValueArgumentLiteImpl.getCached(
+                KSNameImpl.getCached(param.name.asString()),
+                param.getDefaultValue()
+            )
+        }
+}
+
+fun ValueParameterDescriptor.getDefaultValue(): Any? {
```

Comment:
Have you tried `getCompileTimeInitializer`, inherited from `VariableDescriptor`?

Reply:
Yes, it was not working.